app/assets/scripts/components/success-modal.js

```diff
@@ -23,24 +34,47 @@ var SuccessModal = React.createClass({
     if (!this.props.errors && this.props.visible) this.toggleModal();
   },
 
+  uploadData: function () {
+    s3Upload(this.props.csvFile).then((response) => {
+      if (response.status === 201) {
+        this.setState({
+          title: 'Upload Success!',
+          message: <div className='modal__body--about'>
```

Comment:
The state should be used to store only the data that's needed to generate the content. You shouldn't "pre-render" components in the state.
In this case you'd use a `status` flag (like success or failure) and then based on that render different content.